File: baml_language/crates/baml_tests/benches/swe_bench/tasks/greenfield/text-stats/reference/python/text_stats.py
"""Reference implementation. Used to validate the grader before any
agent runs. Place this as text_stats.py in the staging dir to confirm
all fixtures pass."""

import json
import sys
from pathlib import Path


def main() -> int:
    if len(sys.argv) != 2:
        print("usage: text_stats.py <path>", file=sys.stderr)
        return 2
    raw = Path(sys.argv[1]).read_bytes()
    text = raw.decode("utf-8")
    out = {
        "bytes": len(raw),
        "chars": len(text),
        "words": len(text.split()),
        "lines": text.count("\n"),
    }
    print(json.dumps(out, separators=(",", ":")))
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: baml_language/crates/baml_tests/benches/swe_bench/tasks/greenfield/text-stats/reference/go/text_stats.go
// Reference implementation for the text-stats task in Go.
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"strings"
	"unicode/utf8"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Fprintln(os.Stderr, "usage: text_stats <path>")
		os.Exit(2)
	}
	raw, err := os.ReadFile(os.Args[1])
	if err != nil {
		fmt.Fprintf(os.Stderr, "read: %v\n", err)
		os.Exit(2)
	}
	text := string(raw)
	out := map[string]int{
		"bytes": len(raw),
		"chars": utf8.RuneCountInString(text),
		"words": len(strings.Fields(text)),
		"lines": strings.Count(text, "\n"),
	}
	enc, _ := json.Marshal(out)
	fmt.Println(string(enc))
}
